operators/src/processing/raster_vector_join/points.rs

```diff
@@ -1,139 +1,252 @@
+
+        let collection_stream = raster_query
+            .time_multi_fold(
+                move || Ok(PointRasterJoiner::new()),
+                move |accum, raster| {
+                    let points = points.clone();
+                    async move {
+                        let accum = accum?;
+                        let raster = raster?;
+                        accum.extract_raster_values(&points, &raster)
+                    }
+                },
+            )
+            .map(move |accum| accum?.into_colletion(new_column_name));
 
-            let mut rasters = raster_processor.raster_query(query, ctx)?;
+        collection_stream.boxed()
+    }
+}
 
-            // TODO: optimize geo access (only specific tiles, etc.)
+struct PointRasterJoiner<P: Pixel> {
+    values: Vec<Option<P>>,
+    points: Option<MultiPointCollection>,
+}
 
-            while let Some(raster) = rasters.next().await {
-                let raster = raster?;
-                let geo_transform = raster.tile_information().tile_geo_transform();
+impl<P: Pixel> PointRasterJoiner<P> {
+    fn new() -> Self {
+        // TODO: is it possible to do the initialization here?
 
-                for feature_index in time_span.feature_index_start..=time_span.feature_index_end {
-                    // TODO: don't do random access but use a single iterator
-                    let geometry = points.geometry_at(feature_index).expect("must exist");
+        Self {
+            values: vec![],
+            points: None,
+        }
+    }
 
-                    // TODO: aggregate multiple extracted values for one multi point before inserting it to the aggregator
-                    for coordinate in geometry.points() {
-                        let grid_idx = geo_transform.coordinate_to_grid_idx_2d(*coordinate);
+    fn initialize(
+        &mut self,
+        points: &MultiPointCollection,
+        raster_time: &TimeInterval,
+    ) -> Result<()> {
+        // TODO: could be paralellized
+
+        let (indexes, time_intervals): (Vec<_>, Vec<_>) = points
+            .time_intervals()
+            .iter()
+            .enumerate()
+            .filter_map(|(i, time)| {
+                time.intersect(raster_time)
+                    .map(|time_intersection| (i, time_intersection))
+            })
+            .unzip();
 
-                        // try to get the pixel if the coordinate is within the current tile
-                        if let Ok(pixel) = raster.get_at_grid_index(grid_idx) {
-                            // finally, attach value to feature
+        let mut valid = vec![false; points.len()];
+        for i in indexes {
+            valid[i] = true;
+        }
 
-                            let is_no_data = raster
-                                .no_data_value()
-                                .map_or(false, |no_data| pixel == no_data);
+        let points = points.filter(valid)?;
+        let points = points.replace_time(&time_intervals)?;
```

Comment:
You basically make an intersection between the validity of the feature and the raster and that is the new validity of the feature for this chunk. For the second question, since this is an intersection, it cannot get larger.